src/librustc_mir/interpret/eval_context.rs

```diff
@@ -252,17 +252,22 @@ impl<'a, 'mir, 'tcx: 'mir, M: Machine<'a, 'mir, 'tcx>> InterpretCx<'a, 'mir, 'tc
         &self,
         substs: T,
     ) -> EvalResult<'tcx, T> {
+        if !substs.needs_subst() {
+            return Ok(self.tcx.erase_regions(&substs));
+        }
+        // Stack frames beyond the first stackframe have their own set of substs and it would be
+        // absolutely wrong to use them for any further substitutions. So we just yield the substs
+        // after region erasing and let the caller figure out if that is good enough for them.
+        if self.stack.len() > 1 {
+            return Ok(self.tcx.erase_regions(&substs));
+        }
```

Comment:
New rounds of reviews: I'm pretty sure this needs to be removed.

Reply:
it's gone